server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CompareAction.java

```diff
@@ -101,9 +104,16 @@ public void handle(Request request, Response response) throws Exception {
     String leftKey = request.mandatoryParam(PARAM_LEFT_KEY);
     String rightKey = request.mandatoryParam(PARAM_RIGHT_KEY);
 
-    QProfileComparisonResult result = comparator.compare(leftKey, rightKey);
-
     try (DbSession dbSession = dbClient.openSession(false)) {
+      QualityProfileDto left = dbClient.qualityProfileDao().selectByKey(dbSession, leftKey);
+      checkArgument(left != null, "Could not find left profile '%s'", leftKey);
+      QualityProfileDto right = dbClient.qualityProfileDao().selectByKey(dbSession, rightKey);
+      checkArgument(right != null, "Could not find right profile '%s'", rightKey);
+
+      checkArgument(Objects.equals(left.getOrganizationUuid(), right.getOrganizationUuid()), "Cannot compare quality profiles of different organizations. Quality profile left with key '%s' belongs to organization '%s', quality profile right with key '%s' belongs to organization '%s'.", leftKey, left.getOrganizationUuid(), rightKey, right.getOrganizationUuid());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Split this 363 characters long line (which is greater than 180 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)